lib/_element.js

```diff
@@ -344,6 +352,70 @@ function element () {
     }))
   }
 
+  Element.prototype.compareDocumentPosition = function (other) {
+    function getFirstNodeByOrder (nodes, nodeOne, nodeTwo) {
+      return nodes.reduce(function (result, node) {
+        return result !== false
+          ? result
+          : node === nodeOne
+            ? nodeOne
+            : node === nodeTwo
+              ? nodeTwo
+              : node.childNodes
+                ? getFirstNodeByOrder(node.childNodes, nodeOne, nodeTwo)
+                : false
+      }, false)
```

Comment:
If only JavaScript's regular if statements were all expressions that automatically returned :sob: you've got a pretty neat solution here though, nice one.